ebos/eclproblem.hh

```diff
@@ -1329,8 +1377,16 @@ public:
                 }
                 std::string msg = "The action: " + action->name() + " evaluated to true at " + ts + " wells: " + wells_string;
                 Opm::OpmLog::info(msg);
-                schedule.applyAction(reportStep, std::chrono::system_clock::from_time_t(simTime), *action, actionResult, {});
+
+                const auto& wellpi = this->fetchWellPI(reportStep, *action, schedule, matching_wells);
+
+                schedule.applyAction(reportStep, std::chrono::system_clock::from_time_t(simTime), *action, actionResult, wellpi);
                 actionState.add_run(*action, simTime);
+
+                for ( const auto& [wname, _] : wellpi) {
+                    (void)_;
+                    this->wellModel_.updateEclWell(reportStep, wname);
+                }
```

Comment:
This looks like it will call `updateEclWell()` even in the case that the actual action is not `WELPI`.  Am I missing something?

Reply:
I'll check

Reply:
Ehh - no the `wellpi` map should only contains wells which are mentioned in the ACTIONX clause in a ActionxX keyword; maybe a better name for `fetchWellPI` would be `fetchWellPI_for_ACTIONX_wells` - although that is a mouthfull. Or do you think I am in the wrong here?

Reply:
> the `wellpi` map should only contains wells which are mentioned in the ACTIONX clause in a ActionxX keyword;

Okay, I guess it comes down to the details.  Suppose we have
```
ACTIONX
A 1 /
WPI PROD01 > 10.0E+3 /
/

WELTARG
PROD01 LRAT 5000 /
/
ENDACTIO

ACTIONX
B 1
WOPR PROD2 < 150 /
/

WELPI
PROD3 400 /
/
ENDACTIO

ACTIONX
C 1 /
WPI PROD4 > 5000 /
/

WELPI
PROD4 4500 /
/
ENDACTIO
```
Which wells will be in `wellpi` if these actions trigger?

Reply:
You don't come up with simple examples do you :grinning:? Anyway I'll give it a shot:

**Action A:** Here the action statement does not contain the `WELPI` keyword and the `wellpi` map will be empty. The fact that the `WPI:PROD01` variable is accessed in the condition clause is irrelevant in this context, that has been evaluated in the summary code and made available in `SummaryState` prior to evaluating actions.

**Action B:** This action mentions the well `PROD3` in the `WELPI` keyword and we will have `wellpi = {"PROD3" : ???}`

**Action C:** This action mentions the well `PRO4` in the `WELPI` keyword and we will have `wellpi = {"PROD4" : xxxx}`

Modulo bugs.

Reply:
@bska : Are you reasonably happy with the answer here?